test/exe/Main.hs

```diff
@@ -1725,13 +1726,77 @@ haddockTests
   where
     checkHaddock s txt = spanDocToMarkdownForTest s @?= txt
 
+
+sessionDepsArePickedUp :: TestTree
+sessionDepsArePickedUp = testSession'
+  "session-deps-are-picked-up"
+  $ \dir -> do
+    liftIO $ do
+      writeFileUTF8 (dir </> "hie.yaml") "cradle: {stack: {component: \"foo:lib\"}}"
+      writeFileUTF8 (dir </> "stack.yaml") stackYamlContent
+      writeFileUTF8 (dir </> "package.yaml") oldPackageYamlContent
+    -- Open without OverloadedStrings and expect an error.
+    doc <- openDoc' "Foo.hs" "haskell" fooContent
+    expectDiagnostics
```

Comment:
I don’t really want to download a stack GHC bindist in a test and this is also timing out on CI. Can we instead use a `direct-style` cradle and test that changes to that are picked up?

Reply:
`direct-style` cradle doesn't work at the moment because it doesn't give any files as the session dependencies. I think it should provide the `hie.yaml` file itself as a dependency but it doesn't currently. Let me see if I can fix that before making changes here.

Reply:
I made a small change in the `cradleToSession` function in `exe/Main.hs` to make this feature work for direct-style cradles. And also updated this test using a direct-style cradle.